server/routes/utils/dataReportHelpers.ts

```diff
@@ -337,6 +343,32 @@ function traverse(data: any, keys: string[], result: { [key: string]: any } = {}
   return result;
 }
 
+/**
+ * Recursively flatten nested objects and arrays.
+ * @param obj The object to flatten.
+ * @param parentKey The parent key for the current level.
+ * @param result The result object to store flattened key-value pairs.
+ */
+function flattenNestedObject(obj: any, parentKey: string, result: { [key: string]: any }) {
+  Object.entries(obj).forEach(([key, value]) => {
+    const newKey = `${parentKey}.${key}`;
+
+    if (Array.isArray(value)) {
+      value.forEach((item, index) => {
+        if (typeof item === 'object' && item !== null) {
+          flattenNestedObject(item, `${newKey}[${index}]`, result);
+        } else {
+          result[`${newKey}[${index}]`] = item;
+        }
+      });
+    } else if (typeof value === 'object' && value !== null) {
```

Comment:
nit. trivial perf but null check is faster, it should be on the left
```suggestion
    } else if (value !== null && typeof value === 'object') {
```

Reply:
this is just one example, there are more occurrences in the code